src/suggestion.cpp

```diff
@@ -139,7 +139,11 @@ Xapian::Query SuggestionDataBase::parseQuery(const std::string& query)
   m_queryParser.set_stemming_strategy(Xapian::QueryParser::STEM_SOME);
   xquery = m_queryParser.parse_query(query, flags);
 
-  if (!query.empty()) {
+  if ( !query.empty() && xquery.get_num_subqueries() == 0 ) {
+    // a non-empty query string produced an empty xapian query which means
+    // that the query string is made solely of punctuation.
```

Comment:
"of punctuation or stop words" ?

Reply:
Currently, stop-words are not used for suggestions.